lib/board.rb

```diff
@@ -27,4 +30,22 @@ def make_free_positions_hash(blanks)
     end
     hash
   end
+
+  def board_state(symbol)
+    return nil if num_of_played_positions < board.size * 2 - 1
+
+    check_rows = express_board_as_rows(board)
+
+    if check_rows.any? { |row| check_uniqueness(row, symbol) }
+      'Winner'
+    elsif num_of_played_positions == board_size * board_size
+      'Tie'
+    end
+  end
+
+  private
+
+  def check_uniqueness(row, mark)
```

Comment:
Another method name I may consider is `win?` since it is a little more intentional.

Reply:
REsolved at 9dbbabed8c5453b180590dc36e5ca3e44f5c06c8